pkg/neorpc/rpcevent/filter.go

```diff
@@ -35,14 +35,22 @@ func Matches(f Comparator, r Container) bool {
 	}
 	switch f.EventID() {
 	case neorpc.BlockEventID:
-		filt := filter.(neorpc.BlockFilter)
+		err := filter.IsValid()
```

Comment:
You don't need to include this check into `Matches`. Instead, add this check to the client-side and to the server-side.

Client-side code should be exactly like in https://github.com/nspcc-dev/neo-go/pull/3258/commits/3840e506719223cb2f7ecc356d2e1446cf2fa908#r1425034760 described:
```
func (c *WSClient) performSubscription(params []any, rcvr notificationReceiver) (string, error) {
	var resp string

	if flt := rcvr.Filter(); flt != nil {
		err := flt.IsValid()
		if err != nil {
			return "", err
		}
	}
```

And the server-side code is OK now.